mozillaspeechlibrary/src/main/java/com/mozilla/speechlibrary/LocalSpeechRecognition.java

```diff
@@ -0,0 +1,341 @@
+             String _ = this.mModel.finishStream(this.mStreamingState);
+        }
+
+        if (this.mModel != null) {
+            Log.e(this._tag, "closeModel()");
+            this.mModel.destroyModel();
+        }
+
+        this.mStreamingState = null;
+        this.mModel          = null;
+    }
+
+    public void appendAudio(short[] aBuffer) {
+        Log.e(this._tag, "appendAudio()");
+        if (!this.stopStream) {
+            // Log.e(this._tag, "appendAudio()::add");
+            this.mBuffers.add(aBuffer);
+
+            if (this.keepClips) {
+                // DEBUG
+                ByteBuffer myByteBuffer = ByteBuffer.allocate(aBuffer.length * 2);
+                myByteBuffer.order(ByteOrder.LITTLE_ENDIAN);
+
+                ShortBuffer myShortBuffer = myByteBuffer.asShortBuffer();
+                myShortBuffer.put(aBuffer);
+
+                try {
+                    this.clipDebug.write(myByteBuffer);
+                } catch (Exception ex) {
+                }
+            }
+        }
+    }
+
+    public void endOfStream() {
+        Log.e(this._tag, "endOfStream()");
+        this.stopStream = true;
+        if (this.keepClips) {
+            try {
+                this.clipDebug.close();
+            } catch (Exception ex) {
+            }
+        }
+    }
+
+    public void run() {
+        Log.e(this._tag, "run()");
+
+        while ((!this.stopStream) || (this.mBuffers.size() > 0)) {
+            short[] aBuffer = this.mBuffers.poll();
+
+            if (aBuffer == null) {
+                continue;
+            }
+
+            this.mModel.feedAudioContent(this.mStreamingState, aBuffer, aBuffer.length);
+        }
+
+        Log.e(this._tag, "finishStream()");
+        mService.notifyListeners(MozillaSpeechService.SpeechState.DECODING, null);
+        String finalDecoded = this.mModel.finishStream(this.mStreamingState);
+        Log.e(this._tag, "finalDecoded(" + finalDecoded.length() + ")=" + finalDecoded);
+        this.mStreamingState = null;
+
+        STTResult sttResult = new STTResult(finalDecoded, (float)(1.0));
+        mService.notifyListeners(MozillaSpeechService.SpeechState.STT_RESULT, sttResult);
+    }
+}
+
+class LocalSpeechRecognition implements Runnable {
+
+    Vad mVad;
+    boolean done;
+    boolean cancelled;
+
+    int mMinimumVoice = 150;
+    int mMaximumSilence = 500;
+    int mUpperLimit = 10;
+
+    static final int FRAME_SIZE = 80;
+
+    int mSampleRate;
+    int mChannels;
+    MozillaSpeechService mService;
+
+    static final String _tag = "LocalSpeechRecognition";
+
+    LocalDSInference mInferer;
+    Thread mInferenceThread;
+
+    public static String kTfLiteModel = "output_graph.tflite";
+    public static String kAlphabet    = "alphabet.txt";
+    public static String kLM          = "lm.binary";
+    public static String kTrie        = "trie";
+
+    private static Map<String,String> mLanguages = new HashMap<String, String>();
+    static {
```

Comment:
To be extended, for now producing DeepSpeech model using the same naming convention for languages as Common Voice, but for example, locale names in MR Browser is not the same. Here is a mapping helping.